spec/swagger_spec.rb

```diff
@@ -1,21 +1,54 @@
 describe "OpenAPI stuff" do
+  let(:rails_routes) do
+    Rails.application.routes.routes.each_with_object([]) do |route, array|
+      r = ActionDispatch::Routing::RouteWrapper.new(route)
+      next if r.internal? # Don't display rails routes
+      next if r.engine? # Don't care right now...
+      next if r.action == "invalid_url_error"
+
+      array << {:verb => r.verb, :path => r.path.split("(").first.sub(/:[_a-z]*id/, ":id")}
+    end
+  end
+
+  let(:swagger_routes) do
+    published_versions = rails_routes.collect do |rails_route|
+      (version_match = rails_route[:path].match(/\/api\/catalog\/v([\d]+\.[\d])\/openapi.json$/)) && version_match[1]
+    end.compact
+    Api::Docs.routes.select do |spec_route|
+      published_versions.include?(spec_route[:path].match(/\/api\/catalog\/v([\d]+\.[\d])\//)[1])
+    end
+  end
+
   describe "Routing" do
     include Rails.application.routes.url_helpers
+    let(:app_name)    { "catalog" }
+    let(:path_prefix) { "/api" }
 
-    context "customizable route prefixes" do
-      before do
-        stub_const("ENV", ENV.to_h.merge("PATH_PREFIX" => random_path, "APP_NAME" => random_path_part))
-        Rails.application.reload_routes!
-      end
+    before do
+      stub_const("ENV", ENV.to_h.merge("PATH_PREFIX" => path_prefix, "APP_NAME" => app_name))
```

Comment:
@syncrou can we use with_modified_env here instead of stubbing constants

Reply:
Seconding replacing the stub with `with_modified_env`. Probably should be moved into an `around` since it seems like most of the specs are relying on env variables and it would be annoying to do `with_modified_env` in every single `it` block.